lib/internal/Magento/Framework/Webapi/ServiceInputProcessor.php

```diff
@@ -519,6 +520,12 @@ public function convertValue($data, $type)
                     foreach ($data as $key => $item) {
                         $result[$key] = $this->_createFromArray($itemType, $item);
                     }
+                } else {
+                    throw new SerializationException(
+                        new Phrase(
+                            'The value "' . $data . '" must be an array.'
+                        )
```

Comment:
Injecting a dynamic value in a translatable string will make it very hard to translate it.
I would suggest doing something like this instead:

`new Phrase('The value "%fieldName" must be an array.', ['fieldName' => $data]);`